experimental/HttpBenchmark/Program.cs

```diff
@@ -0,0 +1,375 @@
+            requestBytes = Encoding.ASCII.GetBytes(requestString);
+            responseLength *= pipeline;
+
+            var threadObjects = new Thread[connections];
+            for (var i = 0; i < connections; i++)
+            {
+                var thread = new Thread(() =>
+                {
+                    var responseBuffer = new byte[responseLength];
+
+                    if (keepalive)
+                    {
+                        using (var socket = CreateSocket(uri).Result)
+                        {
+                            Interlocked.Increment(ref _connections);
+
+                            while (true)
+                            {
+                                SendReceive(socket, requestBytes, responseLength, responseBuffer);
+                                Interlocked.Add(ref _requests, pipeline);
+                            }
+                        }
+                    }
+                    else
+                    {
+                        while (true)
+                        {
+                            using (var socket = CreateSocket(uri).Result)
+                            {
+                                Interlocked.Increment(ref _connections);
+                                SendReceive(socket, requestBytes, responseLength, responseBuffer);
+                                Interlocked.Add(ref _requests, pipeline);
+                            }
+                            Interlocked.Decrement(ref _connections);
+                        }
+                    }
+                });
+                threadObjects[i] = thread;
+            }
+
+            _stopwatch.Start();
+
+            for (var i = 0; i < connections; i++)
+            {
+                threadObjects[i].Start();
+            }
+
+            for (var i = 0; i < connections; i++)
+            {
+                threadObjects[i].Join();
+            }
+
+            return 0;
+        }
+
+        private static bool ResponseComplete(byte[] buffer, int count)
+        {
+            var response = Encoding.ASCII.GetString(buffer, 0, count);
+
+            // Determine if response uses Content-Length or Chunked
+            int? contentLength = null;
+            bool chunked = false;
+            using (var reader = new StringReader(response))
+            {
+                string line;
+                while ((line = reader.ReadLine()) != null)
+                {
+                    if (line.StartsWith("Content-Length: "))
+                    {
+                        contentLength = int.Parse(line.Substring("Content-Length: ".Length));
+                        break;
+                    }
+                    else if (line == "Transfer-Encoding: chunked")
+                    {
+                        chunked = true;
+                        break;
+                    }
+                }
+            }
+
+            if (contentLength != null)
```

Comment:
I updated the tool to verify the status code of the initial response used to calculate the response length.  However, subsequent responses are not yet verified.

Currently, the tool takes a lot of shortcuts and makes a lot of assumptions to achieve high performance.  It executes a single request and does expensive response parsing to determine the response length and verify the status code.  For the load test, it assumes all responses have the same content length, and it doesn't verify the status code.  This would break in a lot of scenarios.  For example, a chunked response can be chunked in many different ways yet all are the same effective response.

A better implementation would be to fully parse every response, handle chunking, etc.  And do this efficiently enough so the client doesn't become a bottleneck for a high-performance server like Kestrel Plaintext.  This can be added if and when this tool becomes popular.